fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/EntityTables.java

```diff
@@ -18,88 +18,114 @@
+import static org.apache.fineract.infrastructure.dataqueries.data.StatusEnum.WITHDRAWN;
+import static org.apache.fineract.infrastructure.dataqueries.data.StatusEnum.WRITE_OFF;
+
 import com.google.common.collect.ImmutableList;
-import java.util.ArrayList;
-import java.util.HashMap;
+import jakarta.validation.constraints.NotNull;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
 public enum EntityTables {
 
-    CLIENT("m_client", ImmutableList.of(StatusEnum.CREATE.getCode(), StatusEnum.ACTIVATE.getCode(), StatusEnum.CLOSE.getCode()),
-            "client_id"), LOAN("m_loan",
-                    ImmutableList.of(StatusEnum.CREATE.getCode(), StatusEnum.APPROVE.getCode(), StatusEnum.DISBURSE.getCode(),
-                            StatusEnum.WITHDRAWN.getCode(), StatusEnum.REJECTED.getCode(), StatusEnum.WRITE_OFF.getCode()),
-                    "loan_id"), GROUP("m_group",
-                            ImmutableList.of(StatusEnum.CREATE.getCode(), StatusEnum.ACTIVATE.getCode(), StatusEnum.CLOSE.getCode()),
-                            "group_id"), SAVING("m_savings_account",
-                                    ImmutableList.of(StatusEnum.CREATE.getCode(), StatusEnum.APPROVE.getCode(),
-                                            StatusEnum.ACTIVATE.getCode(), StatusEnum.WITHDRAWN.getCode(), StatusEnum.REJECTED.getCode(),
-                                            StatusEnum.CLOSE.getCode()),
-                                    "savings_account_id"), SAVINGS_TRANSACTION("m_savings_account_transaction", ImmutableList.of(),
-                                            "savings_account_transcation_id"), OFFICE("m_office", ImmutableList.of(),
-                                                    "office_id"), PRODUCT_LOAN("m_product_loan", ImmutableList.of(),
-                                                            "product_loan_id"), SAVINGS_PRODUCT("m_savings_product", ImmutableList.of(),
-                                                                    "savings_product_id"), SHARE_PRODUCT("m_share_product",
-                                                                            ImmutableList.of(), "share_product_id");
-
-    private static final Map<String, EntityTables> lookup = new HashMap<String, EntityTables>();
-
-    static {
-        for (EntityTables d : EntityTables.values()) {
-            lookup.put(d.getName(), d);
-        }
-    }
+    CLIENT("m_client", "client_id", "id", CREATE, ACTIVATE, CLOSE), //
+    GROUP("m_group", "group_id", "id", CREATE, ACTIVATE, CLOSE), //
+    CENTER("m_center", "m_group", "center_id", "id"), //
+    OFFICE("m_office", "office_id", "id"), //
+    LOAN_PRODUCT("m_product_loan", "product_loan_id", "id"), //
+    LOAN("m_loan", "loan_id", "id", CREATE, APPROVE, DISBURSE, WITHDRAWN, REJECTED, WRITE_OFF), //
+    SAVING_PRODUCT("m_savings_product", "savings_product_id", "id"), //
```

Comment:
Please correct the typos for savings. It should be:
SAVINGS
SAVINGS_PRODUCT
SAVINGS_TRANSACTION

Reply:
I was thinking on this, but SAVING and SAVING_PRODUCT were already here, and if I would change this, then it would mean several other classes to be changed, which I wanted to avoid. So I just followed the naming with SAVING_TRANSACTION.
I can rename of course, please let me know if I should do that and modify the depending classes.

Reply:
It was SAVINGS_PRODUCT before your change :/
I would not change them if not necessary, but if you want to have them on the same naming convention, i am fine at the end.